src/main/java/com/example/customerdb/service/PurchaseInfoServiceImpl.java

```diff
@@ -40,6 +40,11 @@ public PurchaseInfo updateInfo(int id, PurchaseInfo purchaseInfo) {
     @Override
     public PurchaseInfo editInfo(int id, PurchaseInfo purchaseInfo) {
         PurchaseInfo existingInfo = purchaseInfoMapper.findOptionalById(id).orElseThrow(() -> new NotFoundException(id));
+        purchaseInfoMapper.update(id, purchaseInfo);
+        existingInfo.setName(purchaseInfo.getName());
+        existingInfo.setEmail(purchaseInfo.getEmail());
+        existingInfo.setPurchaseDate(purchaseInfo.getPurchaseDate());
+        existingInfo.setPrice(purchaseInfo.getPrice());
```

Comment:
imo
論理的に考えると
```suggestion
        PurchaseInfo existingInfo = purchaseInfoMapper.findOptionalById(id).orElseThrow(() -> new NotFoundException(id));
        existingInfo.setName(purchaseInfo.getName());
        existingInfo.setEmail(purchaseInfo.getEmail());
        existingInfo.setPurchaseDate(purchaseInfo.getPurchaseDate());
        existingInfo.setPrice(purchaseInfo.getPrice());
        purchaseInfoMapper.update(id, existingInfo);
```
というように既存情報を置き換えてそれをupdateするという展開が読み取れて良いかなと思いました。